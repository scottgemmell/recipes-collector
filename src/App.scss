@import "./mixins";



%u-visually-hidden,
.u-visually-hidden {
	display: none;
}

%u-spinner,
.u-spinner {
	// height: 100vh;
	// width: 100vw;
	display: flex;
	justify-content: center;
	align-items: center;
}

body {
	background-color: #f0f0f0;
	margin: 0;
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
	  'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
	  sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}

a,
button {
	color: rgb(112, 76, 182);
	text-decoration: none;

	&:focus {
		border-color: orange;
		outline: none;
		color: orange;
	}
}
  
code {
	font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
		monospace;
}

dt {
	font-weight: bold;
	margin-bottom: 5px;
}

dd {
	margin-bottom: 10px;
}
  
  

  
  
.App-link {
	color: rgb(112, 76, 182);
}


.list {
	display: grid;
	grid-template-columns: 1fr;
	margin: 0 -10px;
	padding: 0;

	@include respond-to('small') {
		grid-template-columns: 1fr 1fr;
	}
	
	@include respond-to('medium') {
		grid-template-columns: 1fr 1fr 1fr;
	}
	
	@include respond-to('large') {
		grid-template-columns: 1fr 1fr 1fr 1fr;
	}
}
  
.thumb {
	background-color: #fff;
	display: flex;
	justify-content: flex-end;
	align-items: center;
	flex-direction: column-reverse;
	margin: 10px;
	border: 5px solid #fff;

	.thumb__img {
		width:100%;
    	max-height:100vw;
		display: block;
		//object-fit: cover;
		img {
			max-width:100%;
			max-height:100%;
			width:auto;
			height:auto;
			display:block;
			margin:0 auto;
		}
	}

	.thumb__meta {
		padding: 14px 10px 6px 10px;
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		text-align: left;
		width: calc(100% - 20px);
	}
	
	button {
		border: none;
		font-size: 20px;
		padding: 0;
		margin-left: 16px;
		margin-right: 6px;
		//color: #704cb6;
		background-color: transparent;

		&:focus {
			outline: 0;
		}
	}
}
  
.App-main {
	padding-bottom: 60px;

	.App-inner {
		  max-width: 800px;
		  margin: 0 auto;
		  padding: 10px;
	}
}

.item {
	display: flex;
	align-items: flex-start;

	.item__thumb {
		img {
			border: 5px solid #fff;
			margin-top: 20px;
			margin-left: 0;

			.item__meta {
				margin-left: 20px;
				margin-right: 20px;
			}
		}
	}
}

.cta {
	border: 2px solid #704cb6;
	border-radius: 5px;
	background: none;
	padding: 14px 12px;
	font-size: 16px;
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	font-weight: bold;
	color: #704cb6;

	svg {
		font-size: 24px;
	}

	span {
		margin-left: 6px;
	}
}
