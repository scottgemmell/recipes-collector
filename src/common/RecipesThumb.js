import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { FaHeart, FaRegHeart } from "react-icons/fa";

const RecipesThumb = ({ idMeal, isFav, strMealThumb, strMeal, handleFavourite }) => {

	return (
		<React.Fragment>
			<li className="thumb">
				<div className="thumb__meta">
					<Link 
						className="thumb__title" 
						to={`/recipe/${idMeal}/`}
					>
						{strMeal}
					</Link> 
					<button 
						className="thumb__toggle" 
						onClick={() => handleFavourite({ idMeal })}
					>
						{isFav 
							? <Fragment>
								<FaHeart /> <span className="u-visually-hidden">Add to favourites</span>
							</Fragment> 
							: <Fragment>
								<FaRegHeart /> <span className="u-visually-hidden">Remove from favourites</span>
							</Fragment>} 
					</button> 
				</div>
				<div className="thumb__img">
					<img 
						src={strMealThumb} 
						width={320} 
						height={320} 
						alt={strMeal} 
					/> 
				</div>
			</li>
		</React.Fragment>
	);
};

RecipesThumb.propTypes = {
	idMeal: PropTypes.string.isRequired,
	isFav: PropTypes.bool.isRequired, 
	strMealThumb: PropTypes.string.isRequired, 
	strMeal: PropTypes.string.isRequired, 
	handleFavourite: PropTypes.func.isRequired,
};

export default RecipesThumb;
