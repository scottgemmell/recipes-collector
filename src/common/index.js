export { default as RecipesList } from "./RecipesList";
export { default as RecipesThumb } from "./RecipesThumb";
export { default as FavouritesThumb } from "./FavouritesThumb";
export { default as FavouritesList } from "./FavouritesList";
export { default as SiteHeader } from "./SiteHeader";
export { default as Spinner } from "./Spinner";