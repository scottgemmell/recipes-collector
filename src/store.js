import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import counterReducer from "./features/counter/counterSlice";
import favouritesReducer from "./features/favourites/favouritesSlice";
import { createLogger } from "redux-logger";

const logger = createLogger({
	collapsed: true,
});

export default configureStore({
	reducer: {
		counter: counterReducer,
		favourites: favouritesReducer,
	},
	middleware: [...getDefaultMiddleware(), logger],
});
