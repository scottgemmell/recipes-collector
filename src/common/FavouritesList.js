import React from "react";
import PropTypes from "prop-types";
import RecipesThumb from "./RecipesThumb";

const FavouritesList = ({ handleFavourite, list, myFavourites }) => {
	return (
		<ul className="list">
			{list.map(item => (<RecipesThumb 
				key={item.idMeal} 
				{...item} 
				isFav={myFavourites.includes(+item.idMeal)}
				handleFavourite={handleFavourite} 
			/>)
			)}
		</ul>
	);
};

FavouritesList.propTypes = {
	handleFavourite: PropTypes.func.isRequired, 
	list: PropTypes.array.isRequired, 
	myFavourites: PropTypes.array,
};

export default FavouritesList;
