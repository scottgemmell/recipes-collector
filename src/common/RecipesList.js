import React from "react";
import PropTypes from "prop-types";
import RecipesThumb from "./RecipesThumb";

const RecipesList = ({ handleFavourite, list, myFavourites }) => {
	return (
		<ul className="list">
			{list.map(item => (<RecipesThumb 
				key={item.idMeal} 
				{...item} 
				isFav={myFavourites.includes(+item.idMeal)}
				handleFavourite={handleFavourite} 
			/>)
			)}
		</ul>
	);
};

RecipesList.propTypes = {
	handleFavourite: PropTypes.func.isRequired, 
	list: PropTypes.array.isRequired, 
	myFavourites: PropTypes.array.isRequired,
};

export default RecipesList;
