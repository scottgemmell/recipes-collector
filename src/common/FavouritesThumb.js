import React, { Fragment, useEffect, useState } from "react";
import PropTypes from "prop-types";
import { useSelector, useDispatch } from "react-redux";
import { selectFavourites, toggleFavourite } from "../features/favourites/favouritesSlice";
import { Link } from "react-router-dom";
import { FaHeart, FaRegHeart } from "react-icons/fa";
import { Spinner } from "../common";

const FavouritesThumb = ({ idx }) => {

	const myFavourites = useSelector(selectFavourites);
	const dispatch = useDispatch();
	const [recipeResource, setRecipeResource] = useState([]);
	const [isLoading, setIsLoading] = useState(true);
	const isFav = myFavourites.includes(+idx);
	
	useEffect(() => {
		//console.log("STARTUP");
		fetch(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${idx}`)
			.then((res) => {
				return res.json();
			})
			.then((data) => {
			//console.log("B", data.meals[0]);
				return data.meals[0];
			})
			.then((data) => {
			//console.log("DATA", data)
				setRecipeResource(data);
				setIsLoading(false);
			});
	}, [idx]);

	const handleFavourite = (e) => {
		//console.log("handleFavourite", e);
		dispatch(toggleFavourite(e.idMeal));
	};

	return (
		<React.Fragment>
			{isLoading 
				? <Spinner /> 
				: <li className="thumb">
					<div className="thumb__meta">
						<Link className="thumb__title" to={`/recipe/${idx}/`}>
							{recipeResource.strMeal}
						</Link> 
						<button 
							className="thumb__toggle" 
							onClick={() => handleFavourite({ idMeal: +idx })}
						>
							{isFav 
								? <Fragment>
									<FaHeart /> <span className="u-visually-hidden">Add to favourites</span>
								</Fragment> 
								: <Fragment>
									<FaRegHeart /> <span className="u-visually-hidden">Remove from favourites</span>
								</Fragment>} 
						</button> 
					</div>
					<div className="thumb__img">
						<img 
							src={recipeResource.strMealThumb} 
							width={320} 
							height={320} 
							alt={recipeResource.strMeal} 
						/> 
					</div>
				</li>}
		</React.Fragment>
	);
};

FavouritesThumb.propTypes = {
	idx: PropTypes.number.isRequired, 
};

export default FavouritesThumb;
