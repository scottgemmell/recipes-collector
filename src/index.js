import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.scss";
import store from "./store";
import { Provider } from "react-redux";
import { ExamplePage, ItemPage, ListPage } from "./pages/";
import { SiteHeader } from "./common/";
import Favourites from "./features/favourites/Favourites";

ReactDOM.render(
	<Provider store={store}>
		<Router>
			<SiteHeader />
			<Switch>
				{/* <Route path="/list/">
					<ListPage />
				</Route> */}
				<Route path="/favourites/">
					<Favourites />
				</Route>
				<Route path="/example/">
					<ExamplePage />
				</Route>
				<Route path={"/recipe/:mealId/"}>
					<ItemPage />
				</Route>
				<Route path={["/", "/list/"]}>
					<ListPage />
				</Route>
			</Switch>
		</Router>
	</Provider>,
	document.getElementById("root")
);
