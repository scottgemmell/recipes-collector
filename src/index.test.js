import React from "react";
import { shallow } from "enzyme";
//import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store";
import { SiteHeader } from "../src/common";

const defaultProps = () => ({});
const setup = (overrideProps = {}) => {
	const props = Object.assign({}, defaultProps(), overrideProps);
	const wrapper = shallow(<Provider store={store}><SiteHeader {...props} /></Provider>);
	return { wrapper, props };
};

describe("<SiteHeader />", () => {
	it("renders", () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
	});
});
