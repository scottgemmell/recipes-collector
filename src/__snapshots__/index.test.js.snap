// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteHeader /> renders 1`] = `ShallowWrapper {}`;
